Fix project name overflow in app header

b12f opened the pull request:
Longer project names were overflowing in the app header, causing the
text to flow behind the logo, search and user menu. This commit fixes
that issue. This also fixes the menu toggle shrinking in width whenever
there was too little space. Instead, the project title is shrunk.

Closes https://community.openproject.org/projects/openproject/work_packages/43314/activity

Commit:
Fix project name overflow in app header

Longer project names were overflowing in the app header, causing the
text to flow behind the logo, search and user menu. This commit fixes
that issue. This also fixes the menu toggle shrinking in width whenever
there was too little space. Instead, the project title is shrunk.

Closes https://community.openproject.org/projects/openproject/work_packages/43314/activity

HDinger reviewed (approved):
The overflow has been fixed correctly 👍 It seems as if on tablet devices there is some space wasted but that might have been like this beforehand. Could you double check that?
<img width="614" alt="Bildschirmfoto 2022-07-27 um 07 47 04" src="https://user-images.githubusercontent.com/7457313/181170896-e08fb2fe-2a56-4dda-86d3-35bde2cfae34.png">

If it is existing behaviour, it is fine to merge.

Commit:
Give header project selector all available space

On mobile and tablet the instance logo dissapears from the app header,
but the left and right side were both still taking 50% of available
space. However, it makes sense that the project selector can use this
screen estate to show longer project names.
